### ISSUE TITLE
Downloader doesn't have the default datasets to skip and so grabs too much

### ISSUE THREAD

**Issue by dougli1sqrd** [geneontology/pipeline#122]
The excluded datasets in the mega-make go-site Makefile have default values, indicating some datasets we never want to process in the makefile. 

The downloader somehow needs this information to appropriately not overzealously download sources. 

Possible solutions are to put the downloading back in the Makefile, but just as a separate step. Another is to move that skipping configuration into the datasets YAML. We should discuss what that could look like. 

For now we are removing the downloader: https://github.com/geneontology/pipeline/issues/120

**Comment by dougli1sqrd** [comment]
Proposal:
Just to clarify above a little more: there are variables in the Jenkinsfile that in non-snapshot runs override default values in the Makefile. But in snapshot runs, we turn off all overrides, since the default is all data -- excepting the _default_ excluded files defined in the makefile. The issue arose when the Jenkinsfile overrides were assumed to be always present, and when they weren't the downloader did not have the Makefile's defaulted excluded datasets, causing the downloader to download them anyway.

So we need a way to communicate that some datasets should always be skipped.

I'm considering a change that would move the default excluded datasets into the various dataset configs. We have two datasets that we skip by default for all branches of the pipeline: `goa_uniprot_gcrp` and `goa_pdb`. Right now this is only defined in the Makefile as an environment variable, and the Jenkinsfile has to also know it to keep in sync. 

This is where the downloader went wrong. Consider some kind of config change in the datasets yaml where we could specify there that these datasets should be skipped by software that is processing them. This would let us control this configuration in a central place that all pipeline work already uses. We could easily see which are skipped by default and change it later if need be. It could look something like this.
```
   id: goa_pdb.gaf
   label: "goa_pdb gaf file"
   description: "gaf file for goa_pdb from EBI Gene Ontology Annotation Database"
   url: http://geneontology.org/gene-associations/goa_pdb.gaf.gz
   type: gaf
   dataset: goa_pdb
   submitter: goa
   compression: gzip
   source: ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/PDB/goa_pdb.gaf.gz
   entity_type: protein
   status: active
   species_code:
   taxa:
   exclude: true
```
Note the `exclude: true`
I think with rather minor changes in ontobio's `validate.py` script and the downloader script we could make them aware of an exclusion status and not process them. 

This seems like a very correct solution. There would be other faster more hacky ways to do this, but would make all of this more brittle in the future.

What do you think, @kltm and @cmungall ?

**Comment by kltm** [comment]
The removal of hidden information into a higher level or config is a Good Thing.
We can control the subsets eliminated using branches for specialized runs.
Given the increasing complexity of the hidden information and the fact that it has been the cause of much hair pulling in the last month, I'd go ahead with this as a prerequisite for the downloader (I'd rephrase as adding a feature).

Could you run through how you envision this working for partial sets like `master`?

**Comment by kltm** [comment]
Talked to @dougli1sqrd and this is more about just getting the Makefile data out, rather than a larger move of metadata to the YAML. Still good, better than what we have now, and will enable the downloads to move forward with being less convoluted.

**Comment by dougli1sqrd** [comment]
These changes are on in go-site, and schema updated.
